From f7a3d2671110c6a27a294461bfff42e0aacb8db8 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 27 Oct 2025 17:23:47 +0100
Subject: [PATCH] dt-bindings: fix warnings on access-controllers property for
 USB

In the stm32mpxxx device tree files, the access-controllers property has
been added to the USB peripheral nodes so that the firewall configuration
is checked before using them. Document it.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I4ba59680e1a1fc9f6a986d13784b98b768d8a47f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/503463
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 Documentation/devicetree/bindings/usb/stm32,dwc3.yaml | 3 +++
 Documentation/devicetree/bindings/usb/stm32,usbh.yaml | 3 +++
 2 files changed, 6 insertions(+)

--- a/Documentation/devicetree/bindings/usb/stm32,dwc3.yaml
+++ b/Documentation/devicetree/bindings/usb/stm32,dwc3.yaml
@@ -47,6 +47,9 @@ properties:
     maxItems: 1
     description: interrupt used for wakeup
 
+  access-controllers:
+    maxItems: 1
+
 # Required child node:
 
 patternProperties:
--- a/Documentation/devicetree/bindings/usb/stm32,usbh.yaml
+++ b/Documentation/devicetree/bindings/usb/stm32,usbh.yaml
@@ -42,6 +42,9 @@ properties:
     maxItems: 1
     description: interrupt used for wakeup
 
+  access-controllers:
+    maxItems: 1
+
 # Required child node:
 
 patternProperties:
